java/ql/src/Violations of Best Practice/Implementation Hiding/GetClassGetResource.ql

```diff
@@ -12,22 +12,15 @@
 
 import java
 
-/** Access to a method in `this` object. */
-class MethodAccessInThis extends MethodAccess {
-  MethodAccessInThis() {
-    not this.hasQualifier() or
-    this.getQualifier() instanceof ThisAccess
-  }
-}
-
-from Class c, MethodAccess getResource, MethodAccessInThis getClass
+from Class c, MethodAccess getResource, MethodAccess getClass
 where
   getResource.getNumArgument() = 1 and
   (
     getResource.getMethod().hasName("getResource") or
     getResource.getMethod().hasName("getResourceAsStream")
   ) and
   getResource.getQualifier() = getClass and
+  getClass.isOwnMethodAccess() and
```

Comment:
This is actually a behavior change because it now also matches `super.getClass()`, but that is fine because the method returns the runtime type, so `this.getClass() == super.getClass()`.